### ISSUE TITLE
Poetry run no longer executes local bash scripts

### ISSUE THREAD

**Issue by oefe** [python-poetry/poetry#7959]
- **Poetry version**: 1.5.0
- **Python version**: 3.11.3
- **OS version and name**:  macOS 13.2.1 (22D68)
- **pyproject.toml**: [Basic pyproject.toml as created by `poetry new poetry-bug`](https://gist.github.com/oefe/1c8f02fa54de74b7f68807d024e5035d)

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

In Poetry 1.4.2, you could execute local shell scripts using `poetry run`, e.g. `poetry run ./pyright.sh`.

This no longer works with Poetry 1.5

```
>ls -l *.sh
-rwxr-xr-x  1 user  staff  1871 May 19 17:49 pyright.sh*

>head -n 1 pyright.sh
#!/bin/bash

>poetry ./pyright.sh
The command "./pyright.sh" does not exist.

>poetry pyright.sh
The command "./pyright.sh" does not exist.
```

The changelog mentions #7606, but this should only affect python scripts listed in `[tool.poetry.scripts]`, which is not the case here.

So I would expect that this still works.

## Workaround

Explicitly calling `bash` with the script as argument, i.e. `poetry run bash ./pyright.sh` still works

**Comment by bellini666** [comment]
Having the same issue for django projects here. They usually have a `manage.py` executable and in my case, I have a lot of scripts that does `poetry run ./manage.py ...` which are now failing

**Comment by rowanseymour** [comment]
@bellini666 same here, have updated them to `poetry run python manage.py ...` which works

**Comment by danielhoherd** [comment]
I'm having the same issue for python scripts.
```
$ echo 'print("Hello world")' > test.py
$ chmod +x test.py
$ poetry run ./test.py
Command not found: ./test.py
```
however, specifying the interpreter does work:
```
$ echo 'print("Hello world")' > test.py
$ poetry run python ./test.py
Hello world
$ echo 'echo "Hello from bash"' > test.sh
$ poetry run bash test.sh
Hello from bash
```

**Comment by danielhoherd** [comment]
Verified that this was fixed in poetry 1.5.1
